errors.go

```diff
@@ -106,6 +106,24 @@ func New(message string) error {
 	}
 }
 
+// ExistStack return an error already with stack
+// ExistStack will check all parent error
+func ExistStack(err error) bool {
+	type stackTracer interface {
+		StackTrace() StackTrace
+	}
+	if _, ok := err.(stackTracer); ok {
+		return true
+	}
+	type causer interface {
+		Cause() error
+	}
+	if value, ok := err.(causer); ok {
+		return ExistStack(err.Cause())
+	} else {
+		return false
```

Comment:
Every code path in the preceding `if … { … }` block returns from the function, therefore there is no reason to use an `else` block.

Considering using:
```
cause, ok := err.(causer)
if !ok {
  return false
}
return ExistStack(cause.Cause())
```

Please check the implementation of `Cause(err error)` below, and try and mirror that as much as possible.

Reply:
Thank you very much for your help